hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/contract/ContractTestUtils.java

```diff
@@ -1115,8 +1122,9 @@ public static void validateVectoredReadResult(List<FileRange> fileRanges, byte[]
     for (FileRange res : fileRanges) {
       CompletableFuture<ByteBuffer> data = res.getData();
       try {
-        ByteBuffer buffer = FutureIOSupport.awaitFuture(data);
-        assertDatasetEquals((int) res.getOffset(), "vecRead", buffer, res.getLength(), DATASET);
+        ByteBuffer buffer = FutureIO.awaitFuture(data);
+        assertDatasetEquals((int) res.getOffset(), "vecRead",
+                buffer, res.getLength(), originalData);
       } catch (Exception ex) {
         LOG.error("Exception while running vectored read ", ex);
```

Comment:
remove this clause and just throw up all exceptions. look at `awaitFuture` to see what it raises